lib/iris/_data_manager.py

```diff
@@ -219,15 +239,20 @@ def data(self, data):
             managed.
 
         """
+        # If data is None, ensure previous shape is maintained, and that it is
+        # not wrapped in an np.array
+        dataless = data is None
+        if dataless:
+            self._shape = self.shape
```

Comment:
@ESadek-MO Don't we require to set `self._real_array = None` and `self._lazy_array = None` then check the axiom **only** in the `dataless` case?

i.e., do that and nothing further.